### PR TITLE
Annotate exported function return types.

### PR DESCRIPTION
**Pull request by NTillmann**
Release notes: None

I manually reviewed and annotated all exported function return types.
Still TODO: Same for class methods, and then we need some enforcement mechanism.

### CODE REVIEW

File: src/evaluators/ForStatement.js [hunk 1]
```diff
@@ -11,7 +11,13 @@
 
 import type { LexicalEnvironment } from "../environment.js";
 import type { Realm } from "../realm.js";
-import { AbstractValue, Value, EmptyValue, ECMAScriptSourceFunctionValue } from "../values/index.js";
+import {
+  AbstractValue,
+  Value,
+  EmptyValue,
+  ECMAScriptSourceFunctionValue,
+  type UndefinedValue,
```
Comment (hermanventer) [inline src/evaluators/ForStatement.js]: Why "type"?

Comment (NTillmann) [reply src/evaluators/ForStatement.js]: Because we only need the type definition for Flow. No need to pull in the actual value.

Comment (hermanventer) [reply src/evaluators/ForStatement.js]: Does that make Flow faster?

Comment (NTillmann) [reply src/evaluators/ForStatement.js]: Not Flow, but possibly runtime.

Comment (hermanventer) [reply src/evaluators/ForStatement.js]: In the context of Prepack, that seems a bit unlikely to me. Startup is not our problem.

**Pull request closed without merging**